src/blockMonitor/blockProcessor.ts

```diff
@@ -1,27 +1,62 @@
 import { ethers } from "ethers";
 import { StartStopService, ApplicationError } from "../dataEntities";
 import { ReadOnlyBlockCache, BlockCache } from "./blockCache";
-import { IBlockStub } from "./blockStub";
+import { IBlockStub } from "../dataEntities";
+import { Block, HasTxHashes } from "../dataEntities/block";
+
+type BlockFactory<T> = (provider: ethers.providers.Provider) => (blockNumberOrHash: number | string) => Promise<T>;
+
+export const minimalBlockFactory = (provider: ethers.providers.Provider) => async (
```

Comment:
I would call this blockStubAndTxFactory (even though that's not quite true). Since minimal isn't quite true either,

Reply:
Both of these factories should have some documentation about what they return

Reply:
Done.